core/src/components/button/button.tsx

```diff
@@ -33,10 +33,7 @@ export class TdsButton {
   connectedCallback() {
     this.slotElement = this.host.children ? this.host.children[0] : null;
     if (this.disabled) {
```

Comment:
What happens if we move this one to componentWillRender? That way it will run on every render, and we can check if we disabled === true and do what we need.